daffodil-runtime1/src/main/scala/org/apache/daffodil/udf/UserDefinedFunctionService.scala

```diff
@@ -226,13 +228,35 @@ object UserDefinedFunctionService extends Logging {
           val maybeUdf =
             try {
               val udf = udfInfo.provider.createUserDefinedFunction(namespaceURI, fname)
-              Option(udf)
+              /*
+               * This is to check for any errors thrown when if we try to serialize the
+               * UDF, such as when using save-parser
+               */
+              try {
+                new ObjectOutputStream(new ByteArrayOutputStream()).writeObject(udf)
+                Option(udf)
+              } catch {
+                case e: NotSerializableException =>
+                  log(LogLevel.Error, "Error serializing initialized User Defined Function: %s. Could not serialize member of class: %s",
+                    udf.getClass.getName, e.getMessage)
+                  None
+              }
             } catch {
+              /*
+             * This is to protect against any errors thrown when we are trying to
+             * initialize the UDFs. It will catch any exceptions and emit them as the reason the UDF
+             * is unsupported
+             */
               case e: ReflectiveOperationException => {
                 log(LogLevel.Error, "Error initializing User Defined Function: %s. Error thrown: %s",
                   udfid, e.getCause)
                 None
               }
+              case e: Exception => {
```

Comment:
In general we try to avoid catching a general exception, but it doesn't seem unreasonable to have a generic one here. Are there other exceptions you know of that were being hit, or is this just to be safe?

If we do want to keep the generic Exception case, can we remove the ReflectiveOperationException since Exception should handle that? Looks like the only different is ``e.getCause`` vs just ``e``.

Reply:
The addition of this generic exception is mainly to be safe as the ReflectiveOperationException catches errors thrown by our default UDFP createUDF function (and as these tend the wrap the underlying issue, we call a getCause on the e). 

We don't catch errors caused by the improper programming of a UDFP implementation such as a scala.MatchError (which returned a huge ugly "This is a bug" message in my case; though on 2nd thought, this might not be fully necessary as the error message does identify the UDF createUDF as the culprit...so I can go either way). Since the exception is not wrapped in anything, we need to pass it directly to log.

If we only want to krrp generic, I could refactor it to check if e is a ReflectiveOperationException within the log call, and call a e.getCause if it is.

Reply:
Let's say the serialization code for a UDF throws a divide-by-zero arithmetic exception. 

What do we want to happen? I think we want a Daffodil UDF-oriented exception object with a cause that is the arithmetic exception.  I believe we should not just log this. We need to let the exception propagate. 

The "this is a bug" banner is appropriate for this situation. Some of these "User" defined functions will be ours. Their failure really would be a bug. 

Usually, if you are catching very general things like Exception, you can repackage things into other exception objects, but you should end by re-throwing something.